capture/crawler_estadao.py

```diff
@@ -0,0 +1,247 @@
+
+def extract_published_time(url, soup):
+	""" Get the news published datetime 	
+
+	:param soup: object with news html page
+	:type soup: BeautifulSoup object
+	:return: news published datetime
+	:rtype: string
+
+	"""
+
+	MONTHS = {	u"janeiro": u"Jan", u"fevereiro": u"Fev", u"mar\xe7o": u"Mar", u"abril": u"Apr",
+				u"maio": u"May", u"junho": u"Jun", u"julho": u"Jul", u"agosto": u"Aug",
+				u"setembro": u"Sep", u"outubro": u"Oct", u"novembro": u"Nov", u"dezembro": u"Dec"
+			 }
+
+	try:
+		time = soup.findAll("p", {"class":"data"})[0].text
+	except:
+		try:
+			time = soup.findAll("span", {"class":"data"})[0].text
+		except:
+			logger.error('wrong data tags')
+			return None
+	
+	try:
+		time = time.strip().split()
+		time[1] = time[1].lower()
+		time[1] = MONTHS[time[1]]
+		
+		if url.find("estadao.com.br/noticias/") is not -1:
+			time = time[0:3] + time[4:6]
+			time[3] = time[3][0:2]
+			
+		elif url.find("estadao.com.br/blogs/") is not -1:
+			time[3] = time[4][0:2]
+			time[4] = time[4][3:5]
+	except:
+		logger.error('wrong data extraction')
+		return None
+	
+	time = '-'.join(time)
+	
+	try:
+		published_time = datetime.datetime.strptime(time, '%d-%b-%Y-%H-%M')
+		return published_time
+	except:
+		logger.error('wrong published time format')
+		return
+
+def extract_title(soup):
+	""" Extract the news title.
+
+	"""
+	
+	try:
+		title = soup.findAll('h1', {'class':'titulo'})[0].text
+	except:
+		try:
+			title = soup.findAll('h2', {'class':'subtitulo'})[0].text
+		except:
+			logger.error('wrong title tag or attribute')
+			return
+
+	return title
+
+def extract_content(url, response_content):
+	""" Extract relevant information about news page
+
+	:param url: news page's url
+	:param response_content:
+	:type url: string
+	:type response_content:
+	:return: compressed content, title, published time and body content of news page 
+	:rtype: dict()
+
+	"""
+	
+	cleaned_content = clean_content(response_content)
+	
+	if url.find("estadao.com.br/noticias/") is not -1:
+		soupy = Soupy(cleaned_content).find("div", {"itemprop":"articleBody"})
+		try:
+			content = soupy.children.each(Q.text.strip()).filter(len).val()
+		except:
+			try:
+				soupy = Soupy(cleaned_content).find("article")
+				content = soupy.children.each(Q.text.strip()).filter(len).val()
+			except:
+				logger.error("wrong content tags or attributes")
+				return
+
+	elif url.find("estadao.com.br/blogs/") is not -1:
+		try:
+			soupy = Soupy(cleaned_content).find("article")
+			content = soupy.children.each(Q.text.strip()).filter(len).val()
+		except:
+			logger.error("wrong tags or attributes")
+			return
+
+	article = " ".join(content) 
+	return article
+	
+def download_article(url):
+	""" Download the html content of a news page
+
+	:param url: news page's url
+	:type url: string
+	:return: news page's content 
+	:rtype: requests.models.Response
+
+	"""
+
+	article = {
+		'link': url,
+		'source': 'crawler_estadao'
+	}
+
+	logger.info("Downloading article: %s", url)
```

Comment:
I'd be in favor of using only new style string formatting, since we're on the way to migrate everything to python3 and also because not mixing both styles make it easier to read the entire code.